src/methods/abstract.js

```diff
@@ -107,24 +107,34 @@ export function AbstractRelationalComparison(
   realm: Realm,
   x: ConcreteValue,
   y: ConcreteValue,
-  LeftFirst: boolean
-): BooleanValue | UndefinedValue {
+  LeftFirst: boolean,
+  op?: BabelBinaryOperator
+): BooleanValue | UndefinedValue | AbstractValue {
   let px, py;
 
   // 1. If the LeftFirst flag is true, then
   if (LeftFirst) {
     // a. Let px be ? ToPrimitive(x, hint Number).
-    px = To.ToPrimitive(realm, x, "number");
+    px = To.ToPrimitiveOrAbstract(realm, x, "number");
 
     // b. Let py be ? ToPrimitive(y, hint Number).
-    py = To.ToPrimitive(realm, y, "number");
+    py = To.ToPrimitiveOrAbstract(realm, y, "number");
   } else {
     // 2. Else the order of evaluation needs to be reversed to preserve left to right evaluation
     // a. Let py be ? ToPrimitive(y, hint Number).
-    py = To.ToPrimitive(realm, y, "number");
+    py = To.ToPrimitiveOrAbstract(realm, y, "number");
 
     // b. Let px be ? ToPrimitive(x, hint Number).
-    px = To.ToPrimitive(realm, x, "number");
+    px = To.ToPrimitiveOrAbstract(realm, x, "number");
+  }
+
+  if (px instanceof AbstractValue || py instanceof AbstractValue) {
+    invariant(op !== undefined, "operator must be provided if abstract values are to be compared");
```

Comment:
It seems that caller's will have a hard time ensuring that this invariant is never violated, unless they always pass in op. So, shouldn't op be non optional?

Reply:
I was not sure of touching `./intrinsics/ecma262/ArrayPrototype.js`

https://github.com/facebook/prepack/blob/master/src/intrinsics/ecma262/ArrayPrototype.js#L1353-L1354

https://github.com/facebook/prepack/blob/master/src/intrinsics/ecma262/ArrayPrototype.js#L1357-L1358

Is it okay update the call over there?

Reply:
Yes. The operator use would be "<". The call sites should also assert that their return values are BooleanValue instances.